.gitignore

```diff
@@ -21,3 +21,9 @@ hs_err_pid*
 *.iml
 **/target
 
+.project
+.settings
+.cache-main
+.cache-tests
+.classpath
+*/.gitignore
```

Comment:
why */.gitignore?

Reply:
it seems to me that there is no need for a separate gitignore file under each sub folder/project so I add this line. If that is not true or you want more flexibility, of course we could remove this line